gen/builders/generator.go

```diff
@@ -218,67 +244,91 @@ func (g *Generator) Group(group string, vectors ...*VectorDef) {
+				for _, v := range variants {
+					var (
+						tmp      = vectorPath(tmpDir, group, &item, v)
+						existing = vectorPath(g.OutputPath, group, &item, v)
+					)
 
-				if g.OutputPath != "" {
-					outFile := vectorPath(g.OutputPath, group, item)
-					_, err := os.Stat(outFile)
-					exists := !os.IsNotExist(err)
+					out, err := os.OpenFile(tmp, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+					if err != nil {
+						log.Printf("failed to open file for writing %s: %s", tmp, err)
+						continue
+					}
 
-					// if file (probably) exists and we're not force overwriting it, check equality
-					if exists && g.Mode != OverwriteForce {
-						eql, err := g.vectorsEqual(tmpFile, outFile)
+					if err := json.NewEncoder(out).Encode(v); err != nil {
+						log.Printf("failed to write json into file %s: %s", tmp, err)
+						continue
+					}
+
+					_ = out.Close()
+
+					switch _, err := os.Stat(existing); {
+					case err == nil:
+						// file exists.
+						if g.Mode == OverwriteForce {
+							// skip straight to the writing.
+							break
+						}
+						eql, err := g.vectorsEqual(tmp, existing)
 						if err != nil {
 							log.Printf("failed to check new vs existing vector equality: %s", err)
-							return
+							continue
 						}
 						if eql {
-							log.Printf("not writing %s: no changes", item.Metadata.ID)
-							return
+							log.Printf("not writing %s: no changes", existing)
+							continue
 						}
 						if g.Mode == OverwriteNone {
-							log.Printf("⚠️ WARNING: not writing %s: vector changed, use -u or -f to overwrite", item.Metadata.ID)
-							return
+							// no overwrite requested, warn that the vector has changed but we're refusing to overwrite.
+							log.Printf("⚠️ WARNING: not writing %s: vector changed, use -u or -f to overwrite", existing)
+							continue
+						}
+						if g.Mode == OverwriteUpdate {
+							// acknowledge that the vector has changed and we will overwrite
+							log.Printf("test vector exists, is not equal, and update was requested, overwriting: %s", existing)
 						}
+					case os.IsNotExist(err):
+						// file doesn't exist, write it.
+					default:
+						log.Printf("failed unexpectedly while checking if file exists: %s; err: %s", existing, err)
+						continue
 					}
+
 					// Move vector from tmp dir to final location
-					if err := os.Rename(tmpFile, outFile); err != nil {
+					if err := os.Rename(tmp, existing); err != nil {
 						log.Printf("failed to move generated test vector: %s", err)
 					}
+
 					// If this vector was broken and became fixed, then remove the broken
 					// vector file (and vice versa).
-					if err := removePrevious(g.OutputPath, group, item); err != nil {
+					if err := removePrevious(g.OutputPath, group, &item, v); err != nil {
 						log.Printf("failed to remove previously broken vector: %s", err)
 					}
-					log.Printf("wrote test vector: %s", outFile)
+					log.Printf("wrote test vector: %s", existing)
 				}
-			}(item)
+
+			}(*item)
```

Comment:
Nitpick, but I am not sure why dereference here, when you have to then take the address of the dereferenced values three times in the go func()... maybe I am missing smth, but previous version seemed cleaner.

Reply:
Because we want to force a copy of the item for better (but not complete) safety!